### PR TITLE
PODS-2490 : Scheme variations - Nav for establishers, directors / partners

### PR DESCRIPTION
**Pull request by abhijeetardale**
PODS-2490 : Scheme variations - Nav for establishers, directors / partners

### CODE REVIEW

File: app/navigators/EstablishersCompanyDirectorNavigator.scala [hunk 1]
```diff
@@ -19,100 +19,132 @@ package navigators
...
+        exitMiniJourney(establisherIndex, directorIndex, mode, srn)
+      case DirectorContactDetailsId(establisherIndex, directorIndex) =>
+        exitMiniJourney(establisherIndex, directorIndex, mode, srn)
       case DirectorAddressYearsId(establisherIndex, directorIndex) =>
-        addressYearsEditRoutes(establisherIndex, directorIndex)(from.userAnswers)
+        addressYearsEditRoutes(establisherIndex, directorIndex, mode, srn)(from.userAnswers)
+      case _ => commonRoutes(from, mode, srn)
+    }
+
+  protected def commonRoutes(from: NavigateFrom, mode: Mode, srn: Option[String]): Option[NavigateTo] =
```
Comment (grantrobson) [inline app/navigators/EstablishersCompanyDirectorNavigator.scala]: Cyclomatic complexity warning here. Either reduce complexity or switch off this check with scalastyle:off

File: app/navigators/EstablishersCompanyDirectorNavigator.scala [hunk 2]
```diff
@@ -19,100 +19,132 @@ package navigators
...
         NavigateTo.dontSave(controllers.routes.SessionExpiredController.onPageLoad())
     }
   }
+
+  private def listOrAnyMoreChange(establisherIndex: Int, mode: Mode, srn: Option[String])(answers: UserAnswers): Option[NavigateTo] = {
+    mode match {
+      case CheckMode | NormalMode =>
+        NavigateTo.save(controllers.register.establishers.company.routes.AddCompanyDirectorsController.onPageLoad(mode, srn, establisherIndex))
+      case _ => answers.get(IsEstablisherNewId(establisherIndex)) match {
+        case Some(true) if(mode == CheckMode || mode == NormalMode) =>
```
Comment (grantrobson) [inline app/navigators/EstablishersCompanyDirectorNavigator.scala]: Unnecessary brackets.

Comment (abhijeetardale) [reply app/navigators/EstablishersCompanyDirectorNavigator.scala]: resolved

File: app/navigators/EstablishersCompanyDirectorNavigator.scala (hunk 1 above)
Comment (abhijeetardale) [reply app/navigators/EstablishersCompanyDirectorNavigator.scala]: unable to find it

**Pull request merged**